src/types/auth.ts

```diff
@@ -1,32 +1,42 @@
-import { Session } from 'next-auth'
+import {
+	Session as NextAuthSession,
+	Profile as NextAuthProfile,
+} from 'next-auth'
 import { JWT } from 'next-auth/jwt'
 
 /**
  * Object representing an authentication session.
  */
-type SessionData = Omit<Session, 'user'> & {
+export type SessionData = Omit<NextAuthSession, 'user'> & {
 	accessToken?: string
 	id?: string
 	error?: AuthErrors
+	user: NextAuthSession['user'] & {
+		nickname: string
+	}
+}
+
+export type SessionStatus = 'authenticated' | 'unauthenticated' | 'loading'
```

Comment:
Defining this type in our app because I couldn't find it defined in next auth types.